elftools/common/utils.py

```diff
@@ -102,6 +102,41 @@ def roundup(num, bits):
     """
     return (num - 1 | (1 << bits) - 1) + 1
 
+
+class lazy(object):
```

Comment:
I'm not convinced that this magic is really needed, and I would very much prefer to keep the library simple. Lazy parsing of specific constructs can be done with simple code in the specific places where it's required. I left some specific questions in other parts of the code, but the larger design question is - what gets re-parsed unnecessarily and why, and what is the easiest way to prevent that?

Reply:
Yes, lazy parsing with caching can be implemented in any place where it is needed. But it creates some similar code in each the place. I think, it's better to keep such an auxillary code in one place. And then to attach it to a special code using a decorator.